### PR TITLE
Add -s auto-complete option

### PR DESCRIPTION
**Pull request by animaone**
Add -s auto-complete option.

Commented at the following issue: https://github.com/Genymobile/scrcpy/issues/3522

### CODE REVIEW

File: app/data/bash-completion/scrcpy [hunk 1]
```diff
@@ -93,6 +93,11 @@ _scrcpy() {
             COMPREPLY=($(compgen -W 'verbose debug info warn error' -- "$cur"))
             return
             ;;
+        -s|--serial)
+            # Use 'adb devices' to list serial numbers.
+            COMPREPLY=($(compgen -W "$(adb devices | awk '/(device|recovery|sideload)$/{print $1}')" -- ${cur} ) )
```
Comment (rom1v) [inline app/data/bash-completion/scrcpy:98]: I think that scrcpy may not work in recovery or sideload state, so maybe `adb devices |awk '/device$/{print $1}'` is sufficient.

**Review by rom1v — COMMENTED** [review]
Awesome, thank you. :+1:

Could you also add it to the zsh completion script?

File: app/data/bash-completion/scrcpy [hunk 2]
```diff
@@ -95,7 +95,7 @@ _scrcpy() {
             ;;
         -s|--serial)
             # Use 'adb devices' to list serial numbers.
-            COMPREPLY=($(compgen -W "$(adb devices | awk '/(device|recovery|sideload)$/{print $1}')" -- ${cur} ) )
+            COMPREPLY=($(compgen -W "$(adb devices | awk '/device$/{print $1}')" -- ${cur} ) )
```
Comment (rom1v) [inline app/data/bash-completion/scrcpy:98]: Also, if the `ADB` environment variable is set, this adb should be used instead.

```bash
"${ADB:-adb}" devices | awk …
```

**Comment by animaone** [comment]
Ok. I've added a similar solution to the zsh completion. :)

**Comment by rom1v** [comment]
Thank you :+1:

I rebased your commits onto `dev`, and made the following changes:
 - handle the case where the `ADB` environment variable is set
 - slightly adapt the `awk` command: `awk '$2 == "device" {print $1}'`
 - use the same `awk` command for zsh completion

I pushed to branch [`pr3523`](https://github.com/Genymobile/scrcpy/commits/pr3523). Could you please check?
If that's ok, I'll merge into `dev`.

**Comment by animaone** [comment]
Checked. Many thanks +1    :)

**Comment by rom1v** [comment]
Thanks :+1:

Merged into `dev`. :rocket:

**Pull request closed without merging**